src/app/services/article.service.ts

```diff
@@ -1,25 +1,50 @@
-import { Injectable } from '@angular/core';
-import { Article } from "../interfaces/article";
-import { HttpClient } from "@angular/common/http";
-import { delay, switchMap, from, Observable, of, tap } from "rxjs";
-import { getHardcodedArticles, getHardcodedArticle } from './hardcoded.article';
+import {Injectable} from '@angular/core';
+import {Article} from "../interfaces/article";
+import {from, Observable, of, switchMap, take} from "rxjs";
+import {getHardcodedArticle} from './hardcoded.article';
 
 @Injectable({
   providedIn: 'root'
 })
 export class ArticleService {
-  private articles: Article[] | null = [];
+  private articles: Article[] = [];
   private featuredArticles: string[] = [
-    "stories", 
-    "church-construction-complete", 
-    "is-peter-rock", 
+    "stories",
+    "church-construction-complete",
+    "is-peter-rock",
     "why-not-catholic"
   ];
   private defaultImageURI = "https://t4.ftcdn.net/jpg/02/97/78/03/360_F_297780357_pK8VCA7wctbTFusAGiCfcoxbJLRwC9Bs.jpg";
-  constructor(private http: HttpClient) {
+
+  constructor() {
   }
 
-  loadArticles = async () => {
+  public readOne(articleId: string | null): Observable<Article> {
```

Comment:
I'm not a fan of `readOne` name. I'd rather it be called `getArticle`.

Reply:
I can agree. Something like _getArticleById(id)_ may be better